### ISSUE TITLE
catch variables should not be flagged as unused with `--noUnusedParamters`

### ISSUE THREAD

**Issue by mhegazy** [Microsoft/TypeScript#9402]
```ts
        try {
           throw new Error();
        }
        catch (e) {
            return false;
        }
```

`e` should not be an error.

**Comment by glen-84** [comment]
@mhegazy Should this change now that catch clause variables are optional? (and I assume you could also use `_e`)

Error:

```ts
catch (e) { /* `e` not used */ }
```

Okay:

```ts
catch (_e) { /* `e` not used */ }
```

Okay:

```ts
catch { /* `e` not used */ }
```

**Comment by mhegazy** [comment]
we can tighten it up now that we have the optional catch clause. we also need to update the quick fix to handle it correctly. a PR would be appreciated.
